gcloud-java-dns/src/main/java/com/google/gcloud/testing/LocalDnsHelper.java

```diff
@@ -0,0 +1,1426 @@
+    ResourceRecordSet soa = new ResourceRecordSet();
+    soa.setTtl(21600);
+    soa.setName(zone.getDnsName());
+    soa.setRrdatas(ImmutableList.of(
+        // taken from the service
+        "ns-cloud-c1.googledomains.com. cloud-dns-hostmaster.google.com. 0 21600 3600 1209600 312"
+    ));
+    soa.setType("SOA");
+    ResourceRecordSet ns = new ResourceRecordSet();
+    ns.setTtl(21600);
+    ns.setName(zone.getDnsName());
+    ns.setRrdatas(zone.getNameServers());
+    ns.setType("NS");
+    RrsetWrapper nsWrapper = new RrsetWrapper(ns);
+    RrsetWrapper soaWrapper = new RrsetWrapper(soa);
+    ImmutableList<RrsetWrapper> results = ImmutableList.of(nsWrapper, soaWrapper);
+    nsWrapper.setId(getUniqueId(results));
+    soaWrapper.setId(getUniqueId(results));
+    return results;
+  }
+
+  /**
+   * Returns a list of four nameservers randomly chosen from the predefined set.
+   */
+  static List<String> randomNameservers() {
+    ArrayList<String> nameservers = Lists.newArrayList(
+        "dns1.googlecloud.com", "dns2.googlecloud.com", "dns3.googlecloud.com",
+        "dns4.googlecloud.com", "dns5.googlecloud.com", "dns6.googlecloud.com"
+    );
+    while (nameservers.size() != 4) {
+      int index = ID_GENERATOR.nextInt(nameservers.size());
+      nameservers.remove(index);
+    }
+    return nameservers;
+  }
+
+  /**
+   * Returns a hex string id (used for a dns record) unique withing the set of wrappers.
+   */
+  static String getUniqueId(List<RrsetWrapper> wrappers) {
+    TreeSet<String> ids = Sets.newTreeSet(Lists.transform(wrappers,
+        new Function<RrsetWrapper, String>() {
+          @Nullable
+          @Override
+          public String apply(@Nullable RrsetWrapper input) {
+            return input.id() == null ? "null" : input.id();
+          }
+        }));
+    String id;
+    do {
+      id = Long.toHexString(System.currentTimeMillis())
+          + Long.toHexString(Math.abs(ID_GENERATOR.nextLong()));
+      if (!ids.contains(id)) {
+        return id;
+      }
+    } while (ids.contains(id));
+    return id;
+  }
+
+  /**
+   * Tests if a DNS record matches name and type (if provided). Used for filtering.
+   */
+  static boolean matchesCriteria(ResourceRecordSet record, String name, String type) {
+    if (type != null && !record.getType().equals(type)) {
+      return false;
+    }
+    if (name != null && !record.getName().equals(name)) {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Returns a project container. Never returns null because we assume that all projects exists.
+   */
+  ProjectContainer findProject(String projectId) {
+    ProjectContainer defaultProject = createProject(projectId);
+    projects.putIfAbsent(projectId, defaultProject);
+    return projects.get(projectId);
+  }
+
+  /**
+   * Returns a zone container. Returns null if zone does not exist within project.
+   */
+  ZoneContainer findZone(String projectId, String zoneName) {
```

Comment:
Here and elsewhere. I don't like so much the `findXXX` naming, is there a reason to prefer it to `getXXX`?